swaps/swaps_service.go

```diff
@@ -125,254 +183,1057 @@ func (svc *swapsService) EnableAutoSwaps(ctx context.Context, lnClient lnclient.
+		}
+	}
+}
 
-			switch parsedStatus {
-			case boltz.SwapCreated:
-				logger.Logger.WithFields(logrus.Fields{
-					"swap":   swap,
-					"update": update,
-				}).Info("Paying the swap invoice")
-				err := lnClient.SendPaymentProbes(ctx, swap.Invoice)
-				if err != nil {
-					logger.Logger.WithField("swapId", swap.Id).Info("Couldn't probe invoice payment, terminating swap")
-					return err
+func (svc *swapsService) startSwapInListener(swap *db.Swap, boltzWs *boltz.Websocket) {
+	network, err := boltz.ParseChain(svc.cfg.GetNetwork())
+	if err != nil {
+		logger.Logger.WithError(err).WithFields(logrus.Fields{
+			"swapId": swap.SwapId,
+		}).Error("Failed to parse network")
+		return
+	}
+
+	ourKeys, err := svc.keys.GetSwapKey(swap.ID)
+	if err != nil {
+		logger.Logger.WithError(err).WithFields(logrus.Fields{
+			"swapId": swap.SwapId,
+		}).Error("Failed to generate swap child private key")
+		return
+	}
+
+	var serializedTree boltz.SerializedTree
+	if err := json.Unmarshal(swap.SwapTree, &serializedTree); err != nil {
+		logger.Logger.WithError(err).WithFields(logrus.Fields{
+			"swapId": swap.SwapId,
+		}).Error("Failed to unmarshal swap tree")
+		return
+	}
+
+	boltzPubkeyBytes, _ := hex.DecodeString(swap.BoltzPubkey)
+
+	boltzPubKey, err := btcec.ParsePubKey(boltzPubkeyBytes)
+	if err != nil {
+		logger.Logger.WithError(err).WithFields(logrus.Fields{
+			"swapId": swap.SwapId,
+		}).Error("Failed to parse boltz pubkey")
+		return
+	}
+
+	decodedPreimageHash, _ := hex.DecodeString(swap.PaymentHash)
+
+	tree := serializedTree.Deserialize()
+	if err := tree.Init(boltz.CurrencyBtc, false, ourKeys, boltzPubKey); err != nil {
+		logger.Logger.WithError(err).WithFields(logrus.Fields{
+			"swapId": swap.SwapId,
+		}).Error("Failed to initialize swap tree")
+		return
+	}
+
+	if err := tree.Check(boltz.NormalSwap, swap.TimeoutBlockHeight, decodedPreimageHash); err != nil {
+		logger.Logger.WithError(err).WithFields(logrus.Fields{
+			"swapId": swap.SwapId,
+		}).Error("Failed to check swap tree")
+		return
+	}
+
+	if err := tree.CheckAddress(swap.LockupAddress, network, nil); err != nil {
+		logger.Logger.WithError(err).WithFields(logrus.Fields{
+			"swapId": swap.SwapId,
+		}).Error("Failed to check address")
+		return
+	}
+
+	paymentRequest, _ := decodepay.Decodepay(swap.Invoice)
+	amount := uint64(paymentRequest.MSatoshi / 1000)
+
+	for {
+		updatesCh := boltzWs.Updates
+
+		for {
+			select {
+			case <-svc.ctx.Done():
+				logger.Logger.WithError(svc.ctx.Err()).WithFields(logrus.Fields{
+					"swapId": swap.SwapId,
+				}).Error("Swap in context cancelled")
+				return
+			case update, ok := <-updatesCh:
+				if !ok {
+					logger.Logger.WithField("swapId", swap.SwapId).Error("Boltz websocket closed unexpectedly, reconnecting...")
+					if err := boltzWs.Connect(); err != nil {
+						logger.Logger.WithError(err).WithFields(logrus.Fields{
+							"swapId": swap.SwapId,
+						}).Error("Reconnection to Boltz websocket failed")
+						return
```

Comment:
Is it ok that we just return here? should we retry instead?

Reply:
(for both swap directions)

Reply:
Fixed